### PR TITLE
Fixes going from app detailed status view to home page

### PR DESCRIPTION
**Pull request by ajainarayanan**
backbutton-programstatus

### CODE REVIEW

File: cdap-ui/app/features/apps/routes.js [hunk 1]
```diff
@@ -34,18 +34,28 @@ angular.module(PKG.name+'.feature.apps')
       })
         .state('apps.detail.overview', {
           url: '/overview',
-          parent: 'apps.detail',
+          // parent: 'apps.detail',
```
Comment (anwar6953) [inline cdap-ui/app/features/apps/routes.js]: just remove if not needed?

**Comment by elfenheart** [comment]
This change actually breaks everything related to apps view...

**Comment by elfenheart** [comment]
There is also the back button from dataset and streams overview page...

**Comment by ajainarayanan** [comment]
@elfenheart Fixed all references of apps and fixed the same issue with datasets and streams features.

For datasets the implementation is a little different where if it is explorable then we show all tabs and if its not we show only programs tab. All other references to a dataset will still have status but the 'datasets.details' state determines if its explorable or not and shows appropriate things.
backbutton

**Comment by elfenheart** [comment]
:+1: LGTM

**Pull request merged**